src/feature/hs/hs_client.c

```diff
@@ -956,6 +956,37 @@ client_get_random_intro(const ed25519_public_key_t *service_pk)
   return ei;
 }
 
+/** Return true iff all intro points for the given service have timed out. */
+static bool
+intro_points_all_timed_out(const ed25519_public_key_t *service_pk)
+{
+  bool ret = false;
+
+  tor_assert(service_pk);
+
+  const hs_descriptor_t *desc = hs_cache_lookup_as_client(service_pk);
```

Comment:
Let's check that we actually got desc != NULL here before dereferencing it.

Reply:
Fixed, see latest commit.